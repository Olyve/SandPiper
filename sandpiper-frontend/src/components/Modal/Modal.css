.modal {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, .25);
}

.modal-popup {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 250px;
  width: 350px;
  background-color: #fff;
  border-radius: 10px;
}

.modal-title {
  text-align: center;
}

.modal-message {
  text-align: center;
  margin-top: 0;
  margin-bottom: 30px;
}

.modal-buttons {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.modal-confirm {
  border: none;
  border-radius: 5px;
  padding: 5px 10px;
  background-color: #2ecc34;
  color: #fff;
  font-size: 14px;
}

.modal-confirm:active {
  background-color: #26a529;
}

.modal-confirm:focus {
  outline: none;
}

.hidden {
  display: none;
}