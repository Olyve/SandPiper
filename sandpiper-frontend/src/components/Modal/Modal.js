import React, { Component } from 'react';
import { connect } from 'react-redux';
import cx from 'classnames';
import './Modal.css';

export class Modal extends Component {
  render() {
    const messageClasses = cx({
      'modal-message': true,
      'hidden': !this.props.showMessage
    });

    const buttonClasses = cx({
      'modal-buttons': true,
      'hidden': !this.props.showButtons
    });

    return (
      <div className='modal'>
        <div className='modal-popup'>
          <div>
            <h3 className='modal-title'>{this.props.title}</h3>
          </div>
          <div>
            <p className={messageClasses}>
              {this.props.message}
            </p>
          </div>
          <div className={buttonClasses}>
            <button autoFocus className='modal-confirm' onClick={this.props.handleConfirm}>
              Okay
            </button>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { ...state.modal }
}

export default connect(mapStateToProps)(Modal);