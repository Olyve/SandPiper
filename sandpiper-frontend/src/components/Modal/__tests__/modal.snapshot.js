import React from 'react';
import { Modal } from '../Modal';

describe('Modal', () => {
  it('should render without crashing', () => {
    let options = {
      isVisible: true,
      title: '',
      showMessage: true,
      message: '',
      showButtons: true
    };
    const wrapper = shallow(<Modal {...options} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should hide message when showMessage is false', () => {
    let options = {
      isVisible: true,
      title: '',
      showMessage: false,
      message: '',
      showButtons: true
    };
    const wrapper = shallow(<Modal {...options} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should hide buttons when showButtons is false', () => {
    let options = {
      isVisible: true,
      title: '',
      showMessage: true,
      message: '',
      showButtons: false
    };
    const wrapper = shallow(<Modal {...options} />);

    expect(wrapper).toMatchSnapshot();
  });
});