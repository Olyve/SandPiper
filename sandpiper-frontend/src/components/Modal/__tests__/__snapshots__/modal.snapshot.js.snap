// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should hide buttons when showButtons is false 1`] = `
<div
  className="modal"
>
  <div
    className="modal-popup"
  >
    <div>
      <h3
        className="modal-title"
      />
    </div>
    <div>
      <p
        className="modal-message"
      />
    </div>
    <div
      className="modal-buttons hidden"
    >
      <button
        autoFocus={true}
        className="modal-confirm"
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;

exports[`Modal should hide message when showMessage is false 1`] = `
<div
  className="modal"
>
  <div
    className="modal-popup"
  >
    <div>
      <h3
        className="modal-title"
      />
    </div>
    <div>
      <p
        className="modal-message hidden"
      />
    </div>
    <div
      className="modal-buttons"
    >
      <button
        autoFocus={true}
        className="modal-confirm"
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;

exports[`Modal should render without crashing 1`] = `
<div
  className="modal"
>
  <div
    className="modal-popup"
  >
    <div>
      <h3
        className="modal-title"
      />
    </div>
    <div>
      <p
        className="modal-message"
      />
    </div>
    <div
      className="modal-buttons"
    >
      <button
        autoFocus={true}
        className="modal-confirm"
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;
